projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-transient.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * REST API endpoint for editing Jetpack Transients.
+ *
+ * @package automattic/jetpack
+ * @since 9.7.0
+ */
+
+/**
+ * Jetpack transients API.
+ *
+ * @since 9.7.0
+ */
+class WPCOM_REST_API_V2_Endpoint_Transient extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'transients';
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		// DELETE /sites/<blog-id>/transients/$name route.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<name>\w{1,172})',
+			array(
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_transient' ),
+					'permission_callback' => array( $this, 'delete_transient_permissions_check' ),
+					'args'                => array(
+						'name' => array(
+							'description'       => __( 'The name of the transient to delete.', 'jetpack' ),
+							'required'          => true,
+							'type'              => 'string',
+							'sanitize_callback' => 'sanitize_text_field',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Delete transient callback.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 * @return array
+	 */
+	public function delete_transient( \WP_REST_Request $request ) {
+		return array(
+			'success' => delete_transient( $request->get_param( 'name' ) ),
+		);
+	}
+
+	/**
+	 * Check if request has read access.
+	 *
+	 * @return bool
+	 */
+	public function delete_transient_permissions_check() {
+		return current_user_can( 'read' );
```

Comment:
If this is going to be called in relation to a user option changing, should we increase this to something like `edit` access?

Reply:
That makes sense, should it be `edit_posts`? I think `edit_users` [is admin access](https://wordpress.org/support/article/roles-and-capabilities/#edit_users).

Reply:
If this is changed to a higher capability, subscriber roles would not be able to delete their own transients.

`edit_users` is an administrator-level capability, see https://wordpress.org/support/article/roles-and-capabilities/#capability-vs-role-table

Reply:
At the same time it means that subscribers can delete any transient. Which _should_ not be a problem since it should be assumed a transient doesn't exist when working with it.

Maybe still worth checking with a SIRT rep?

Reply:
> Which should not be a problem since it should be assumed a transient doesn't exist when working with it.

Yes, I had a thought along these lines, but it doesn't feel right to have anyone deleting any transient they like via the API.

> subscriber roles would not be able to delete their own transients

I don't think the user will be calling this endpoint directly. We'll be making the request as part of the async job to clear the transients on the user's behalf, for their Jetpack sites. In which case we could use the `request_as_blog` call, to check that the request is coming from wpcom. I'm not sure what the user role is for those calls, but we could maybe make it for administrators and those `request_as_blog` calls?

Reply:
I think this endpoint should be available even for Simple sites. In case we wouldn't want, I think we should call one of the site-type helpers to detect and deny the response here.